### ISSUE TITLE
Dialog components' text is not read out loud by JAWS

### ISSUE THREAD

**Issue by Kiefer-Co-D2L** [BrightspaceUI/core#2300]
The `d2l-dialog-confirm` component does not consistently have its text read out loud by JAWS when opened.  Neither does the `d2l-dialog` component with [`describe-content`](https://github.com/BrightspaceUI/core/pull/2030) set.

Both of these dialogs behave correctly with NVDA, the issue appears the be JAWS-specific.

**Comment by Kiefer-Co-D2L** [comment]
Some additional context can be found in this thread on Slack:
https://d2l.slack.com/archives/C0PHG3QB0/p1649430622032239
